search/services/index/__init__.py

```diff
@@ -323,24 +341,21 @@ def bulk_add_documents(self, documents: List[Document],
             actions = ({
                 '_index': self.index,
                 '_type': self.doc_type,
-                '_id': document.id,
+                '_id': document['id'],
                 '_source': asdict(document)
```

Comment:
This is slightly confusing, since this `asdict` from `search.domain` is meant to operate on a `dataclass`

Reply:
Ups, this is actually a bigger problem -- looks like I missed some bits.

Reply:
Oh wait, not as bad as I thought; ``TypedDict`` classes can be instantiated. But need to fix the ``asdict()`` part

Reply:
Ok, fixed in 43579d55587e0b6abf33655d05ba88bc1de8579f